src/plugins/bfetch/common/buffer/tests/timed_item_buffer.test.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TimedItemBuffer } from '../timed_item_buffer';
+import { runItemBufferTests } from './run_item_buffer_tests';
+import { ItemBuffer } from '../item_buffer';
+
+describe('TimedItemBuffer', () => {
+  runItemBufferTests((TimedItemBuffer as unknown) as new () => ItemBuffer<any>);
```

Comment:
This seems odd to me... Is there any way to avoid all of these `as` casts? I get that we're just running the same tests with `TimedItemBuffer` but this just doesn't sit right.

Reply:
Yes, I made `maxItemAge` optional in `TimeItemBuffer` and now we don't need these casts.